test/langtools/tools/javac/processing/model/util/types/TestAnnotationStripping.java

```diff
@@ -0,0 +1,146 @@
+
+            for (ExecutableElement m : methodsIn(hostClassElt.getEnclosedElements())) {
+                TypeMirror returnType = m.getReturnType();
+
+                System.err.println("Checking " + returnType);
+
+
+                checkExpectedTypeAnnotations(returnType, expectedAnnotation);
+
+//                 System.err.print("\tasElement()");
+//                 checkEmptyAnnotations(typeUtils.asElement(returnType));
+
+                System.err.print("\tcapture()");
+                checkEmptyAnnotations(typeUtils.capture(returnType));
+
+                System.err.print("\terasure()");
+                checkEmptyAnnotations(typeUtils.erasure(returnType));
+
+//                 System.err.print("\tgetArrayType()");
+//                 checkEmptyAnnotations(typeUtils.getArrayType(returnType));
+
+                // System.out.println(returnType.getAnnotation(TypeAnnotation.class));
+                // System.out.println(returnType.getAnnotationsByType(TypeAnnotation.class).length);
+                // TypeAnnotation ta = requireNonNull(returnType.getAnnotation(TypeAnnotation.class));
+
+                System.err.println();
+                System.err.println();
+            }
+
+            if (failures > 0)
+                throw new RuntimeException(failures + " failures occured.");
+        }
+        return true;
+    }
+
+    private int failures = 0;
+
+    void checkExpectedTypeAnnotations(AnnotatedConstruct ac, TypeMirror expectedAnnotation) {
+        List<? extends AnnotationMirror> annotations = ac.getAnnotationMirrors();
+        if (annotations.size() != 1) {
+            failures++;
+            System.err.println("\t\t\tUnexpected annotations size: " + annotations);
+        }
+        if (!typeUtils.isSameType(annotations.get(0).getAnnotationType(), expectedAnnotation)) {
+            failures++;
+            System.err.println("\t\t\tUnexpected annotations type: " + annotations);
+        }
+    }
+
+    void checkEmptyAnnotations(AnnotatedConstruct ac) {
+        System.err.println("\t" + ac);
+        if (ac == null)
+            return;
+        else {
+            List<? extends AnnotationMirror> annotations = ac.getAnnotationMirrors();
+            if (annotations.size() != 0) {
+                failures++;
+                System.err.println(ac.getClass());
+                System.err.println("\t\t\tUnexpected nonzero annotations size: " + annotations);
+            }
+        }
+    }
+}
+
+/*
+ * Class to host annotations for testing
+ */
+class HostClass {
+    public static @TypeAnnotation("foo") Integer foo() {return null;}
+
+    public static @TypeAnnotation("foo2") int foo2() {return 0;}
+
+    public static @TypeAnnotation("foo3") String foo3() {return null;}
+
+    // public static @TypeAnnotation("foo4") java.util.Set foo4() {return null;}
+
+    // public static @TypeAnnotation("foo4") String[] foo4() {return null;}
```

Comment:
It would be good to have a test verifying the annotation on the array component is removed by `erasure`.